builders/interact/ethereum-api/precompiles/batch.md

```diff
@@ -0,0 +1,267 @@
+5. For the value input, since `SimpleContract.sol` does not require any native currency to be paid to it, insert `[0,0]` for 0 Wei
+6. For the **callData** input, insert your call data from the previous section in the following format: `["INSERT_FIRST_CALL_DATA","INSERT_SECOND_CALL_DATA"]`
+7. For the **gasLimit** input, insert `[]`. You can put in a gas limit value for each subcall, or leave it as an empty array
+8. Press **transact**
+9. Press **Confirm** in the MetaMask extension to confirm the transaction
+
+![Batch Function Interaction](/images/builders/interact/ethereum-api/precompiles/batch/batch-6.webp)
+
+If you used the same call data as the tutorial, you can check to make sure that the transaction has been successful as follows:
+
+1. Expand the `SimpleContract.sol` contract under **Deployed Contracts**
+2. To the right of the **messages** button, insert `1`
+3. Press the blue **messages** button
+
+![SimpleContract Confirmation](/images/builders/interact/ethereum-api/precompiles/batch/batch-7.webp)
+
+The phrase **"tanssi"** should appear underneath it. You repeat the above steps with an id of "2" and you should see **"hello"**. Congratulations! You have interacted with a function with the batch precompile.
+
+### Combining Subtransactions {: combining-subtransactions }
+
+So far, transferring native currency and interacting with functions have been separate, but they can be intertwined.
+
+The following four strings can be combined as inputs for a batch transaction. They will transact 1 DEV to the public Gerald (`0x6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b`) account, and interact with a predeployed `SimpleContract.sol` contract twice. Here is a break-down:
+
+There are three subtransactions, so there are three addresses in the `to` input array. The first is the public Gerald account, the next two are a predeployed `SimpleContract.sol` contract. You can replace the last two with your own instance of `SimpleContract.sol` if you wish. Or, replace only one: you can interact with multiple contracts in a single message.
+
+```text
+[
+  "0x6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b",
+  "0xd14b70a55F6cBAc06d4FA49b99be0370D0e1BD39", 
+  "0xd14b70a55F6cBAc06d4FA49b99be0370D0e1BD39"
+]
+```
+
+There will also be three values for the `value` array. The first address in the `to` input array has to do with sending 1 DEV, so 1 DEV in Wei is within the array. The following two values are 0 because the function that their subtransactions are interacting with do not accept or require native currency.  
+
+```text
+["1000000000000000000", "0", "0"]
+```
+
+You will need three values for the `callData` array. Since transferring native currency does not require call data, the string is simply blank. The second and third values in the array correspond to invocations of **setMessage** that set messages to ids 5 and 6.
```

Comment:
```suggestion
You will need three values for the `callData` array. Since transferring native currency does not require call data, the string is simply blank. The second and third values in the array correspond to invocations of **setMessage** that set messages to IDs 5 and 6.
```